providers/nitro/nitro-helper/src/key_utils.rs

```diff
@@ -39,21 +43,57 @@ pub(crate) mod credential {
         }
     }
 
-    /// Gets the role name to get credentials for using the IAM Metadata Service (169.254.169.254).
+    /// get credentials from Aws Instance Metadata Service Version 2
+    /// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html
     pub fn get_credentials() -> Result<AwsCredentials, String> {
+        let token_address = format!("http://{}/{}/", AWS_CREDENTIALS_PROVIDER_IP, AWS_TOKEN_PATH);
+        let client = Client::new();
+        let header_key = HeaderName::from_str("X-aws-ec2-metadata-token-ttl-seconds")
```

Comment:
would error every be returned in `HeaderName::from_str` for this constant?

Reply:
no, before I use `unwrap`, I'll return it back.